src/gluonts/itertools.py

```diff
@@ -289,6 +289,43 @@ def __iter__(self):
         return filter(self.fn, self.iterable)
 
 
+@dataclass
+class MixIterators:
+    iterators: List[Iterator]
+    probabilities: Optional[List[float]] = None
+    random_state: np.random.RandomState = np.random.RandomState()
+
+    def __post_init__(self):
+        if self.probabilities:
+            # copy the probabilities as we may modify it
+            self.probabilities = [prob for prob in self.probabilities]
+        else:
+            self.probabilities = [1.0 / len(self.iterators)] * len(
+                self.iterators
+            )
+
+    def __iter__(self):
```

Comment:
I think we can just do everything in `__iter__`, no need for `__next__`.

Reply:
This is related to the `Iterable` or `Iterator` question. If we stick to `Iterator`, then I think it makes sense to implement `__next__` method. So you pass in a list of `Iterator`s, you get a `Iterator` out.